sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -57,8 +57,47 @@ static int max98373_hw_params(struct snd_pcm_substream *substream,
 	return 0;
 }
 
+static int max98373_trigger(struct snd_pcm_substream *substream, int cmd)
+{
+	struct snd_soc_pcm_runtime *rtd = substream->private_data;
+	struct snd_soc_dai *codec_dai;
+	int j, ret = 0;
+	char *pin_name;
+
+	for_each_rtd_codec_dais(rtd, j, codec_dai) {
+		struct snd_soc_component *component = codec_dai->component;
+		struct snd_soc_dapm_context *dapm =
+				snd_soc_component_get_dapm(component);
+
+		pin_name = devm_kasprintf(rtd->dev, GFP_KERNEL, "%s Spk",
+					codec_dai->component->name_prefix);
+
+		switch (cmd) {
+		case SNDRV_PCM_TRIGGER_START:
+		case SNDRV_PCM_TRIGGER_RESUME:
+		case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
+			ret = snd_soc_dapm_enable_pin(dapm, pin_name);
+			if (!ret)
+				snd_soc_dapm_sync(dapm);
+			break;
+		case SNDRV_PCM_TRIGGER_STOP:
+		case SNDRV_PCM_TRIGGER_SUSPEND:
+		case SNDRV_PCM_TRIGGER_PAUSE_PUSH:
+			if (codec_dai->active != 1)
+				break;
```

Comment:
can you explain why the codec_dai would not be active when those cases occur? Is this test necessary?

Reply:
During DSM, this is one of the sequence:
1) start playback
2)start a record stream(IV data)
3) record ends
4)playback ends.
pb & capture stream are on same SSP.
once the capture stream stops, the TRIGGER_STOP will be called for the capture and it will disable_pin if this condition check is not there.
This will cause no audio on playback stream even though it is running.
Hence the change

Reply:
@RDharageswari So is "codec_dai->active==2" in this case? So this would essential ensure pin is disabled when the last codec dai user does trigger-stop, right?

Reply:
YES

Reply:
can you please add the explanation as a comment before the check?

Reply:
Sure